src/index.ts

```diff
@@ -1,81 +1,61 @@
 import * as WebSocket from 'ws';
-import KaiEventEmitter from './KaiEventEmitter';
-
-let ws: WebSocket;
-
-ws.on('open', () => {
-	// TODO:
-	// Test compartibility
-	// Send capabilities
-});
-
-ws.on('message', (data) => {
-	var input = JSON.parse(data.toString());
-	KaiEvents.emit('data', input);
-
-	if(input.success != true) {
-		KaiEvents.emit('error', {
-			errorCode: input.errorCode,
-			error: input.error,
-			message: input.message
-		});
-		return;
-	}
-
-	switch(input.type) {
-		case "gestureData":
-			KaiEvents.emit('gestureData', input.gesture);
-			break;
-		// TODO other types of data
+import { stringify } from 'querystring';
+import * as sdk from './sdk';
+import { print } from 'util';
+import { type } from 'os';
+import { resolve } from 'path';
+import { rejects } from 'assert';
+import * as Promise from 'bluebird';
+let ws:WebSocket;
+var isConnected:boolean = false
+var isAuthenticated:boolean = false
+
+
+connect('123','qwerty')
+
+//connect and authenticate
+function connect(moduleID:string,moduleSecret:string){
+	ws = new WebSocket('ws://localhost:2203')
+	var authToken:sdk.Request = {
+		type:'authentication',
+		moduleId:moduleID,
+		moduleSecret:moduleSecret
 	}
-});
-
-export function connect() {
-	ws = new WebSocket('ws://localhost:2203');
-}
-
-export function subscribe(...capabilities: KaiCapabilities[]) {
-	return new Promise((resolve, reject) => {
-		let json: any = {
-			type: 'setCapabilities'
-		};
-		capabilities.forEach(capability => {
-			json[toCamelCase(capability)] = true;
-		});
-		if(ws) {
-			resolve();
-		}
-		ws.send(json, err => {
-			if(err)
-				reject(err);
-			else
-				resolve();
-		});
-	});
+	
+	ws.on('open',()=>{
+		ws.send(JSON.stringify(authToken,null,'\t'))
+		
+	})
+	recieve().then((message)=>{
+		console.log(message)
+	}).catch((error)=>{
+		console.log(error)
+	}).catch((errorMessage)=>{
+		console.log(errorMessage)
+	})		
 }
 
-export function unsubscribe(...capabilities: KaiCapabilities[]) {
-	return new Promise((resolve, reject) => {
-		let json: any = {
-			type: 'setCapabilities'
-		};
-		capabilities.forEach(capability => {
-			json[toCamelCase(capability)] = false;
-		});
-		if(!ws) {
-			resolve();
-		}
-		ws.send(json, err => {
-			if(err)
-				reject(err);
-			else
-				resolve();
-		});
-	});
+//Capabilities
+
+// function setCapabilities(kaiId){
+// 	var capabilityRequest:sdk.Request
+// 	capabilityRequest.type='setCapabilities'
+// 	capabilityRequest.kaiId = kaiId
+// 	capabilityRequest.
+// }
+
+//Recieve : 
+
+function recieve(){
+	return new Promise((resolve,reject)=>{
+		ws.on('message',(data)=>{
+			var res:sdk.Response = JSON.parse(JSON.parse(JSON.stringify(data)))
```

Comment:
Why are there 2 json.parse()?

Reply:
![Screenshot 2019-04-08 at 11 10 09 PM](https://user-images.githubusercontent.com/21238991/55744866-8e6ffc80-5a53-11e9-988c-82c2078e26dc.png)

Reply:
Hold on a minute, so you're doing `JSON.parse(JSON.stringify(data))`. Wouldn't that just give you data directly? What's the point of the `JSON.stringify()` there?

Reply:
![Screenshot 2019-04-09 at 9 07 20 AM](https://user-images.githubusercontent.com/21238991/55771862-4ed30000-5aa7-11e9-8760-69f34091ce71.png)

Reply:
`data` is an object. You wouldn't need to parse it, if I'm not wrong. `JSON.parse` takes a string only. That might be what's causing the issue